File: C/exercice5_1.c
#include <stdio.h>
#include <pthread.h>
#include <stdlib.h>
#include <sys/types.h>
#include <sys/stat.h>
#include <time.h>
#include <unistd.h>

void *ma_fonction_thread(void *arg)
{
	int *tableau = (int *)arg;
	int n = tableau[0] ;
	printf("affichage des contenus de tableau passé en paramètre\n");
	for (int i = 1; i <=n; ++i)
	{
		printf("%d ", tableau[i]);
	}
	printf("\n");
	pthread_exit(NULL); 
}

int main(int argc, char const *argv[])
{
	int ret ;
	int tableau[] = {5, 2, 4, 6, 8, 10} ;
	pthread_t my_thread ;
	ret = pthread_create(&my_thread, NULL, ma_fonction_thread, (void*)tableau) ;
	if (ret != 0)
	{
		fprintf(stderr, "erreur de création du thread\n");
		exit(1) ;
	}
	pthread_join(my_thread, NULL) ;
	return 0;
}

File: Python/exercice5_1.py
import threading
import time

def ma_fonction_thread(tableau):
    n = tableau[0]
    print("affichage des contenus de tableau passé en paramètre")
    
    for i in range(1, n + 1):
        print(f"{tableau[i]} ", end="")
    print()

def main():
    tableau = [5, 2, 4, 6, 8, 10]
    
    my_thread = threading.Thread(target=ma_fonction_thread, args=(tableau,))
    
    my_thread.start()
    
    my_thread.join()

if __name__ == "__main__":
    main()